nautobot_device_lifecycle_mgmt/jobs/model_migration.py

```diff
@@ -1064,154 +1064,71 @@ def migrate_content_type_references_to_new_model(self, old_ct, new_ct, hide_chan
-                    destination_type=old_ct, relationship=relationship
-                ).exclude(relationship__key__in=excluded_relationships):
-                    self.logger.info(
-                        "The Relationship Association __%s__ will be migrated to Core model __%s__",
-                        html.escape(str(relationship_association)),
-                        html.escape(str(new_ct)),
-                    )
-                    relationship_associations_dlm_destination.append(
-                        {
-                            "relationship_id": relationship_association.relationship_id,
-                            "source_type_id": relationship_association.source_type_id,
-                            "source_id": relationship_association.source_id,
-                            "destination_type_id": relationship_association.destination_type_id,
-                            "destination_id": relationship_association.destination_id,
-                        }
-                    )
-                    relationship_association.delete()
-            except Exception as err:
-                self.logger.warning(
-                    "Encountered exception __%s__ while disassociating instance of Relationship __%s__. Rolling back relationship updates.",
-                    html.escape(str(err)),
-                    html.escape(str(relationship.label)),
+
+            # Duplicate the relationship with the new content type
+            new_relationship = Relationship(
+                key=old_relationship.key,
+                label=old_relationship.label,
+                description=old_relationship.description,
+                type=old_relationship.type,
+                required_on=old_relationship.required_on,
+                source_type=old_relationship.source_type if old_relationship.source_type != old_ct else new_ct,
+                source_label=old_relationship.source_label,
+                source_filter=old_relationship.source_filter,
+                source_hidden=old_relationship.source_hidden,
+                destination_type=old_relationship.destination_type
+                if old_relationship.destination_type != old_ct
+                else new_ct,
+                destination_label=old_relationship.destination_label,
+                destination_filter=old_relationship.destination_filter,
+                destination_hidden=old_relationship.destination_hidden,
+                advanced_ui=old_relationship.advanced_ui,
+            )
+
+            # Rename relationship key to {key}_old, rename label to "{label} (old)" and hide the old relationship in the ui
+            while Relationship.objects.filter(key=old_relationship.key).exists():
+                old_relationship.key = f"{old_relationship.key}_old"
+            while Relationship.objects.filter(label=old_relationship.label).exists():
+                old_relationship.label = f"{old_relationship.label} (old)"
```

Comment:
I'm not sure what those `while` loops are doing here. Are they needed?

Reply:
I just removed these to make this job idempotent. It's ready for review now.